tracer/src/Datadog.Trace/Debugger/Instrumentation/AsyncMethodDebuggerInvoker.SingleProbe.cs

```diff
@@ -234,6 +236,24 @@ public static void LogLocal<TLocal>(ref TLocal local, int index, ref AsyncMethod
             asyncState.SnapshotCreator.StartSampling();
         }
 
+        private static bool ProcessNotHoistedMoveNextLocal<TLocal>(ref TLocal local, AsyncMethodDebuggerState asyncState, string localName)
+        {
+            bool isSucceeded;
+            if (TypeExtensions.IsDefaultValue(ref local))
+            {
+                var wrapLocal = new DebuggerSnapshotSerializer.UnreachableLocal(DebuggerSnapshotSerializer.UnreachableLocalReason.NotHoistedLocalInAsyncMethod);
+                var captureInfo = new CaptureInfo<DebuggerSnapshotSerializer.UnreachableLocal>(value: wrapLocal, type: typeof(TLocal), methodState: MethodState.LogLocal, name: localName, memberKind: ScopeMemberKind.Local);
+                isSucceeded = asyncState.ProbeData.Processor.Process(ref captureInfo, asyncState.SnapshotCreator);
+            }
+            else
+            {
+                var captureInfo = new CaptureInfo<TLocal>(value: local, type: typeof(TLocal), methodState: MethodState.LogLocal, name: localName, memberKind: ScopeMemberKind.Local);
+                isSucceeded = asyncState.ProbeData.Processor.Process(ref captureInfo, asyncState.SnapshotCreator);
+            }
+
+            return isSucceeded;
```

Comment:
nitpick: I would get rid of `isSucceeded` and simply consolidate the return to `return asyncState.ProbeData.Processor.Process(ref captureInfo, asyncState.SnapshotCreator);`

Reply:
Could be but then we have two return statements so I prefer to leave it as is.